### PR TITLE
8308656: RISC-V: vstring_compare doesnt manifest usage of all vector registers

### PR DESCRIPTION
**Pull request by VladimirKempik**
Please review this fix. 
vstring_compare instrinsic ( from c2_MacroAssembler_riscv.cpp ) uses vector registers v6 and v7 ( https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/riscv/c2_MacroAssembler_riscv.cpp#L1482 , vstr1 == v4, lmul=4) , but doesn't manifest their usage in riscv_v.ad file.
This fix resolves this situation.
No noticable difference one might see in generated code for now.

Testing: build testing only.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8308656](https://bugs.openjdk.org/browse/JDK-8308656): RISC-V: vstring_compare doesnt manifest usage of all vector registers

### Reviewers
 * [Yanhong Zhu](https://openjdk.org/census#yzhu) (@yhzhu20 - Author)
 * [Fei Yang](https://openjdk.org/census#fyang) (@RealFYang - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/14102/head:pull/14102` \
`$ git checkout pull/14102`

Update a local copy of the PR: \
`$ git checkout pull/14102` \
`$ git pull https://git.openjdk.org/jdk.git pull/14102/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 14102`

View PR using the GUI difftool: \
`$ git pr show -t 14102`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/14102.diff">https://git.openjdk.org/jdk/pull/14102.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/14102#issuecomment-1559638721)

### CODE REVIEW

**Comment by VladimirKempik** [comment]
macos build failure unrelated, No code touched outside of cpu/riscv in this PR

File: src/hotspot/cpu/riscv/riscv_v.ad [hunk 1]
```diff
@@ -2222,13 +2222,14 @@ instruct varray_equalsC(iRegP_R11 ary1, iRegP_R12 ary2, iRegI_R10 result,
 %}
 
 instruct vstring_compareU(iRegP_R11 str1, iRegI_R12 cnt1, iRegP_R13 str2, iRegI_R14 cnt2,
-                          iRegI_R10 result, vReg_V1 v1, vReg_V2 v2, vReg_V3 v3, vReg_V4 v4, vReg_V5 v5,
+                          iRegI_R10 result, vReg_V1 v1, vReg_V2 v2, vReg_V3 v3, vReg_V4 v4,
+                          vReg_V5 v5, vReg_V6 v6, vReg_V7 v7,
                           vRegMask_V0 v0, iRegP_R28 tmp1, iRegL_R29 tmp2)
```
Comment (yhzhu20) [inline src/hotspot/cpu/riscv/riscv_v.ad:2227]: Hi,
When StrCompNode is `StrIntrinsicNode::UU` or `StrIntrinsicNode::LL`,  if-branch (element_compare) will be executed in `C2_MacroAssembler::string_compare_v`, and `lmul` is set to 2, so v6 and v7 are not used in `string_compareL` and `string_compareU`, here is the code:
C2_MacroAssembler::string_compare_v
```
  if (str1_isL == str2_isL) { // LL or UU
    element_compare(str1, str2, zr, cnt2, tmp1, tmp2, v2, v4, v1, encLL, DIFFERENCE);
    j(DONE);
  } else { // LU or UL
    Register strL = encLU ? str1 : str2;
    Register strU = encLU ? str2 : str1;
```
C2_MacroAssembler::element_compare
```
 bind(loop);
  vsetvli(tmp1, cnt, sew, Assembler::m2);
  vlex_v(vr1, a1, sew);
  vlex_v(vr2, a2, sew);
```

Comment (VladimirKempik) [reply src/hotspot/cpu/riscv/riscv_v.ad]: Right, thank you

**Review by yhzhu20 — APPROVED** [review]
Looks good.

**Comment by zifeihan** [comment]
Hi, I tested tier1 and `test/jdk/jdk/incubator/vector` before and after using this PR , and the test results are consistent, no new errors are introduced.

**Comment by VladimirKempik** [comment]
> Hi, I tested tier1 and `test/jdk/jdk/incubator/vector` before and after using this PR and the test results are the same, no new errors were introduced (using QEMU and with UseRVV enabled in JDK).

Thanks a lot, these vector changes aren't easy to test.

**Pull request closed without merging**